.github/workflows/release.yaml

```
@@ -0,0 +1,32 @@
+name: Release
+
+on:
+  push:
+    branches: [main]
+    tags: ["v*"]
```

I think the event you should use is [release](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release):

```suggestion
    release:
      types: [published]
```